aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -182,6 +189,15 @@ module aptos_framework::transaction_context {
         payload.entry_function_payload
     }
 
+    /// Returns the hash of the current raw transaction, which is used for transaction authentication.
+    /// This function calls an internal native function to retrieve the raw transaction hash.
+    /// In the case of a fee payer transaction, the hash value is generated using 0x0 as the fee payer address.
+    public fun raw_transaction_hash(): vector<u8> {
+        assert!(features::transaction_context_hash_function_update_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_HASH_FUNCTION_UPDATE_NOT_ENABLED));
+        raw_transaction_hash_internal()
+    }
+    native fun raw_transaction_hash_internal(): vector<u8>;
```

Comment:
@igor-aptos , this is an entirely new native function (unlike `get_txn_hash()`). Does this pattern seem right to you, having a public function wrapper and being protected by a feature gate?

Reply:
yes this is correct.

you can register raw_transaction_hash as native as well, so in next release you can remove   raw_transaction_hash_internal completely